src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker_IVsSolutionLoadEvents.cs

```diff
@@ -51,6 +95,217 @@ int IVsSolutionLoadEvents.OnAfterBackgroundSolutionLoadComplete()
+                projectInfo.CommandLineArguments,
+                projectDirectory,
+                isInteractive: false,
+                sdkDirectory: RuntimeEnvironment.GetRuntimeDirectory());
+
+            // TODO: Should come from sln file?
+            var projectName = Path.GetFileNameWithoutExtension(projectFilename);
+            var projectId = GetOrCreateProjectIdForPath(projectFilename, projectName);
+
+            AbstractProject project;
+            if (_projectMap.TryGetValue(projectId, out project))
+            {
+                return project as IWorkspaceProjectContext;
+            }
+
+            OutputToOutputWindow($"\tCreating '{projectName}':\t{commandLineArguments.SourceFiles.Length} source files,\t{commandLineArguments.MetadataReferences.Length} references.");
+            var solution5 = _serviceProvider.GetService(typeof(SVsSolution)) as IVsSolution5;
+            var projectGuid = solution5.GetGuidOfProjectFile(projectFilename);
+            var projectContext = workspaceProjectContextFactory.CreateProjectContext(
+                languageName,
+                projectName,
+                projectFilename,
+                projectGuid: projectGuid,
+                hierarchy: null,
+                binOutputPath: projectInfo.TargetPath);
+
+            projectContext.SetOptions(projectInfo.CommandLineArguments.Join(" "));
+
+            foreach (var sourceFile in commandLineArguments.SourceFiles)
+            {
+                projectContext.AddSourceFile(sourceFile.Path);
+            }
+
+            foreach (var sourceFile in commandLineArguments.AdditionalFiles)
+            {
+                projectContext.AddAdditionalFile(sourceFile.Path);
+            }
+
+            var addedProjectReferences = new HashSet<string>();
+            foreach (var projectReference in projectInfo.ProjectReferences)
+            {
+                var projectReferencePath = projectReference.Substring("/ProjectReference:".Length);
+                var referencedProject = ImmutableProjects.SingleOrDefault(p => StringComparer.OrdinalIgnoreCase.Equals(p.ProjectFilePath, projectReferencePath));
+                if (referencedProject == null)
+                {
+                    referencedProject = (AbstractProject)CreateProjectFromArgumentsAndReferences(
+                        workspaceProjectContextFactory,
+                        projectReferencePath,
+                        allProjectInfos,
+                        targetPathsToProjectPaths);
+                }
+
+                var referencedProjectContext = referencedProject as IWorkspaceProjectContext;
+                if (referencedProjectContext != null)
+                {
+                    addedProjectReferences.Add(projectReferencePath);
+                    projectContext.AddProjectReference(
+                        referencedProjectContext,
+                        new MetadataReferenceProperties());
```

Comment:
We never had them (they were lost in AnyCode.  Added a TODO about getting them back.